apps/dapp/frontend/components/auth-provider.tsx

```
@@ -31,54 +39,105 @@ const AuthContext = createContext<AuthContextType>({
+      refreshTimerRef.current = null;
+    }
+    const exp = decodeExpiry(accessToken);
+    if (!exp) return;
+    const msUntilExpiry = exp * 1000 - Date.now();
+    const delay = Math.max(msUntilExpiry - 30_000, 5_000); // refresh 30s early
+
+    const attempt = async () => {
+      try {
+        const refreshed = await api.auth.refresh();
+        setToken(refreshed.access_token);
+        scheduleProactiveRefresh(refreshed.access_token);
```

ESLint

[error] Error: Cannot access variable before it is declared

`scheduleProactiveRefresh` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/apps/dapp/frontend/components/auth-provider.tsx:94:9
  92 |         const refreshed = await api.auth.refresh();
  93 |         setToken(refreshed.access_token);
> 94 |         scheduleProactiveRefresh(refreshed.access_token);
     |         ^^^^^^^^^^^^^^^^^^^^^^^^ `scheduleProactiveRefresh` accessed before it is declared
  95 |       } catch (err) {
  96 |         // Only a genuine rejection of the refresh token itself (401/403 —
  97 |         // expired, reused, revoked, device mismatch) means the session is

/home/jailuser/git/apps/dapp/frontend/components/auth-provider.tsx:80:3
   78 |   // so the 5-minute lifetime stays invisible — most page views never hit the
   79 |   // reactive 401-retry path in the API client at all.
>  80 |   const scheduleProactiveRefresh = useCallback((accessToken: string) => {
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  81 |     if (refreshTimerRef.current) {
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  82 |       clearTimeout(refreshTimerRef.current);
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 110 |     refreshTimerRef.current = setTimeout(attempt, delay);
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 111 |   }, [clearSession]);
      | ^^^^^^^^^^^^^^^^^^^^^^ `scheduleProactiveRefresh` is declared here
  112 |
  113 |   // Clear session when wallet disconnects
  114 |   useEffect(() => {

(react-hooks/immutability)